packages/wmr/src/plugins/process-global-plugin.js

```diff
@@ -18,15 +43,13 @@ export default function processGlobalPlugin({ NODE_ENV = 'development', env = {}
 		},
 		transform(code) {
 			const orig = code;
-			// TODO: this should probably use acorn-traverse.
-			code = code.replace(
-				/([(){}&|,;=!]\s*)process\.env\.NODE_ENV\s*([!=]==?)\s*(['"])(.*?)\3/g,
-				(str, before, comparator, quote, value) => {
-					let isMatch = value == NODE_ENV;
-					if (comparator[0] == '!') isMatch = !isMatch;
-					return before + isMatch;
-				}
-			);
+
+			const result = transform(code, {
```

Comment:
It looks like there's already a hook for injecting `import.meta.XX` properties we could use, which nicely avoids the transform step:

```js
	{
		name: 'process-global',
		// etc
		resolveImportMeta(property) {
			if (property === 'env') {
				return processObj;
			}
			return null;
		},
		transform(code) {
			// etc
		}
	};
```

Though this injects the whole process object anywhere `import.meta.env` is used, which isn't awesome. I wonder if we could combine the approaches?

```js
	return {
		name: 'process-global',
		// replace `import.meta.env.foo` with `process.env.foo`
		resolveImportMeta(property) {
			if (property === 'env') {
				return 'process.env';  // maybe '$importmeta.env'?
			}
			return null;
		},
		transform(code) {
			if (code.match(/[^\w$]process\.env/)) {
				code = `import process from '\0builtins:process.js';${code}`;
			}
		}
	};
```

(just thoughts fwiw, I'm not actually sure the transform is that expensive, since it's not doing any AST serialization 👍)

Reply:
Saw that hook, but it's not really useful in our situation as it only runs _after_ `transform()`. Because of that our `process` detection is too late and we'll never add that import.

Reply:
A bit unrelated: I'm actually hoping we'll switch to more ast-type transformations as they are harder to get wrong.

Rollup seems to allow plugins to return the `ast` directly [in a transform hook](https://rollupjs.org/guide/en/#transform). I wonder if this could be a mechanism to ensure that we'd only ever have one AST parse and serialization per file in the long term, despite multiple plugins participating in the transformation.